benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -265,8 +278,18 @@ private void handleRunningExperimentState(String experimentID)
         break;
       }
 
+      runningState = experimentModelDAO.getRunningState(experimentID);
+      logger.info("handleRunningExperimentState: runningState == " + runningState);
+
       // Exit as soon as final state is executed
-      if (experimentState != null && experimentState == TERMINATED) {
+      if (experimentState == TERMINATED) {
+        exit = true;
+      }
+
+      // Exit while waiting for the Faban Manager to notify about the result of the scheduled
+      // trial. This exits before the execution of HANDLE_EXPERIMENT_RESULT
+      else if (runningState == RunningState.HANDLE_TRIAL_RESULT
```

Comment:
Why the code is different compared to https://github.com/benchflow/benchflow/pull/534/files#diff-e320626f292c5fd5e3bf5ea062ab6c90R171?